### PR TITLE
Configurations/15-android.conf: fix implicit __ANDROID_API__ handling.

### PR DESCRIPTION
**Pull request by dot-asm**
03ad7c009e16a233c733098db3169c560142ccd3 failed if one didn't pass
explicit -D__ANDROID_API__=N :-(

### CODE REVIEW

**Comment by petrovr** [comment]
Andy Polyakov wrote:
> 03ad7c009e16a233c733098db3169c560142ccd3 failed if one didn't pass
> explicit -D__ANDROID_API__=N :-(
>
I cannot understand this patch .

-D__ANDROID_API__=N is required only for build with "unified" headers..

-1 for now.

**Comment by dot-asm** [comment]
If you don't specify `-D__ANDROID_API__=N` it's supposed to pick latest available. To do that $api is assigned \* and used as glob. It finds the directory, but never assigns the number to $api. As result `-D__ANDROID_API__=*` was passed :-(

**Comment by petrovr** [comment]
Andy Polyakov wrote:
> If you don't specify `-D__ANDROID_API__=N` it's supposed to pick latest available. 

I'm busy to follow you modification. I hop that soon could follow it.

> To do that $api is assigned \* and used as glob. It finds the directory, but never assigns the number to $api. As result `-D__ANDROID_API__=*` was passed :-(

If build is not "unified" __ANDROID_API__ is defined in one header file 
(apilevel.h ?). So preprocessor flag -D__ANDROID_API__=blabla clash with 
define from this header .


Regards,
Roumen Petrov

**Comment by dot-asm** [comment]
In the essence it's about restoring original behaviour. Originally $api was assigned according to certain algorithm. 03ad7c009e16a233c733098db3169c560142ccd3 broke the algorithm and under some circumstances `-D__ANDROID_API__=*` got passed. This commit restores original $api assignment.

**Comment by petrovr** [comment]
Andy Polyakov wrote:
> In the essence it's about restoring original behaviour. Originally $api was assigned according to certain algorithm. 03ad7c009e16a233c733098db3169c560142ccd3 broke the algorithm and under some circumstances `-D__ANDROID_API__=*` got passed. This commit restores original $api assignment.
> 

Ok. Then modification make sense.

Roumen

**Comment by levitte** [comment]
PING (this is fairly low hanging fruit)

File: Configurations/15-android.conf [hunk 1]
```diff
@@ -92,6 +94,11 @@
                         .  "/$tritools-4.9/prebuilt/$host";
                 $user{CC} = "clang" if ($user{CC} !~ m|clang|);
                 $user{CROSS_COMPILE} = undef;
+                if (which("llvm-ar") =~ m|^$ndk/.*/prebuilt/([^/]+)/|) {
+                    $user{AR} = "llvm-ar";
+                    $user{ARFLAGS} = [ "rs" ];
+                    $user{RANLIB} = ":";
```
Comment (mattcaswell) [inline Configurations/15-android.conf]: Not sure I understand what this last line does?

Comment (levitte) [reply Configurations/15-android.conf]: `:` is the no-op shell command

**Comment by levitte** [comment]
Merged.

master:
ddf2e8c8ccad49cff0d6a0a0b26ca892ec4ce33a  Configurations/15-android.conf: detect NDK llvm-ar.
71f2b3171e8b94b0fbdc7fab2d7ca4ea66edec2b Configurations/15-android.conf: fix implicit `__ANDROID_API__` handling.

1.1.1:
3b1928fe64e36634e09482c721f4d3d0c10047a8 Configurations/15-android.conf: detect NDK llvm-ar.
3453dbde7f0c44bcb97f85ee5291fc4fde0a45b2 Configurations/15-android.conf: fix implicit `__ANDROID_API__` handling.

**Pull request closed without merging**